packages/hdwallet-keepkey/src/binance.ts

```diff
@@ -45,16 +45,22 @@ export async function binanceSignTx(
     );
     if (resp.message_type === Core.Events.FAILURE) throw resp;
 
+    const intRegex = /^(0|([1-9][0-9]*))$/;
+    const outputAmount = message.outputs[0].coins[0].amount.toString();
+    if (!intRegex.test(outputAmount)) throw new Error("output amount should be an integer");
+    const inputAmount = message.inputs[0].coins[0].amount.toString();
+    if (!intRegex.test(inputAmount)) throw new Error("input amount should be an integer");
```

Comment:
```suggestion
    const intRegex = /^(0|([1-9][0-9]*))$/;
    const outputAmount = BigNumber(message.outputs[0].coins[0].amount) // maybe need optional chaining or other validation
    const inputAmount = BigNumber(message.inputs[0].coins[0].amount)
    // BigNumber.prototype.isInteger returns false for NaN values
    if (!outputAmount.isInteger()) throw new Error("Output amount must be an integer")
    if (!inputAmount.isInteger()) throw new Error("Input amount must be an integer")
```

Reply:
pedantic nitpick: optional chaining is the opposite of validation. validation explicitly restricts the set of accepted inputs to a function; optional chaining explicitly expands the set of accepted inputs.

Reply:
In this case it works because `undefined` would return `NaN` and thus be invalided by the `isInteger` check. I would consider that valid value validation in this particular case.

It may be worth looking at doing proper object structure validation at the beginning of the operation in the future.